spec/models/sorn_xml_parser_spec.rb

```diff
@@ -1,16 +1,156 @@
+        </P>
+        <P>DHS/USSS is updating this SORN to:</P>
+        <P>(1) Update the categories of individuals to include individuals who could be in proximity to protected persons or areas secured by USSS;</P>
+        </SUPLINF>
+        HEREDOC
+      end
+
+      it "returns clean string" do
+        expected_content = "I. Background In accordance with the Privacy Act of 1974, 5 U.S.C. 552a, the Department of Homeland Security (DHS) United States Secret Service (USSS) proposes to modify and reissue a current DHS system of records titled, DHS/USSS 004 Protection Information System of Records. Information collected in this  system of records is used to assist USSS in protecting its designated protectees, events, and venues. In doing so, USSS maintains necessary information to implement protective measures and to make protective inquiries concerning individuals who may come into proximity of a protectee, access a protected facility or event, or who have been involved in incidents or events that relate to the protective functions of USSS. Further, USSS ensures this protective information is appropriately managed and accessible to authorized users while employing appropriate safeguards to ensure that information is properly protected in accordance to national security standards. DHS/USSS is updating this SORN to: (1) Update the categories of individuals to include individuals who could be in proximity to protected persons or areas secured by USSS;"
+        expect(parser.get_supplementary_information).to eq expected_content
+      end
+    end
+  end
+
+  describe ".find_section" do
+    it "returns clean string" do
+      expect(parser.find_section("SECURITY")).to eq "<p>Unclassified.</p>"
+    end
+  end
+
+  describe ".get_sections" do
+    it "gets all the sections of the PRIACT tag" do
+      expect(parser.send(:get_sections)).to include "SECURITY CLASSIFICATION:" => "<p>Unclassified.</p>"
+    end
+
+    context "with a rare hash header" do
+      let(:xml) do
+        <<~HEREDOC
+        <PRIACT>
+        <HD SOURCE="HD2">
+          <E T="04">System manager and address:</E>
+        </HD>
+        <P>WHATEVER</P>
+        </PRIACT>
+        HEREDOC
+      end
+
+      it "still gets that section" do
+        expect(parser.send(:get_sections)).to include "System manager and address:" => "<p>WHATEVER</p>"
+      end
+    end
+
+    context "with a beginning P tag" do
+      let(:xml) do
+        <<~HEREDOC
+        <PRIACT>
+        <P>SKIP ME</P>
+        <HD SOURCE="HD2">System manager and address:</HD>
+        <P>WHATEVER</P>
+        </PRIACT>
+        HEREDOC
+      end
+
+      it "skips P tags until an HD is found" do
+        expect { parser.send(:get_sections) }.to_not raise_error
+        expect(parser.send(:get_sections)).to include "System manager and address:" => "<p>WHATEVER</p>"
```

Comment:
would it make sense to have an additional test case where there are multiple p nodes in the doc?